penrose-web/src/engine/Evaluator.ts

```diff
@@ -648,18 +873,80 @@ export const insertExpr = (
   const trans = initTrans;
   let name, field, prop;
   switch (path.tag) {
-    case "FieldPath":
+    case "FieldPath": {
       [name, field] = path.contents;
       // NOTE: this will overwrite existing expressions
       trans.trMap[name.contents][field] = { tag: "FExpr", contents: expr };
       return trans;
-    case "PropertyPath":
+    }
+    case "PropertyPath": {
       // TODO: why do I need to typecast this path? Maybe arrays are not checked properly in TS?
```

Comment:
And the same thing is not happening in the previous case?

Reply:
I don't think that was my TODO, but anyway, I removed the type annotations and it works fine.